asteroid/losses/pit_wrapper.py

```diff
@@ -191,6 +221,169 @@ def best_perm_from_perm_avg_loss(loss_func, est_targets, targets, **kwargs):
+
+    @staticmethod
+    def best_part_mix_it_generalized(loss_func, est_targets, targets, **kwargs):
+        """ Find best partition of the estimated sources that gives
+            the minimum loss for the MixIT training paradigm in [1].
+            Valid only for two mixtures, but those mixtures do not
+            necessarly have to contain the same number of sources.
+            It is allowed the case where one mixture is silent.
+
+        Args:
+            loss_func: function with signature (targets, est_targets, **kwargs)
+                The loss function batch losses from.
+            est_targets: torch.Tensor. Expected shape [batch, nsrc, *].
+                The batch of target estimates.
+            targets: torch.Tensor. Expected shape [batch, nsrc, *].
+                The batch of training targets.
+            **kwargs: additional keyword argument that will be passed to the
+                loss function.
+
+        Returns:
+            tuple:
+                :class:`torch.Tensor`: The loss corresponding to the best
+                permutation of size (batch,).
+
+                :class:`torch.LongTensor`: The indexes of the best permutations.
+
+        References:
+            [1] Scott Wisdom and Efthymios Tzinis and Hakan Erdogan and Ron J Weiss
+            and Kevin Wilson and John R Hershey, "Unsupervised sound separation using
+            mixtures of mixtures." arXiv preprint arXiv:2006.12701 (2020) $
+        """
+
+        # check input dimensions
+        assert est_targets.shape[0] == targets.shape[0]
+        assert est_targets.shape[2] == targets.shape[2]
+
+        # get L, N and M
+        L = targets.shape[1]        # number of mixtures 
+        N = est_targets.shape[1]    # number of estimated sources
+
+        if L != 2:
+            raise ValueError('Works only with two mixtures')
+
+        # Generate all unique partitions of any size from a list lst of
+        # length n. Algorithm recursively distributes items over parts
+        def all_combinations(lst):
+            all_combinations = []
+            for k in range(len(lst) + 1):
+                for c in combinations(lst, k):
+                    rest = [x for x in lst if x not in c]
+                    all_combinations.append([list(c), rest]) 
+            return all_combinations
+
+        # Generate all the possible partitions
+        parts = all_combinations(range(N))    
+        for p, partition in enumerate(parts):
+            assert len(partition) == L
+        
+            # sum the sources according to the given partition
+            est_mixes = torch.stack([torch.sum(est_targets[:, indexes, :], axis=1) for indexes in partition], axis=1)
+
+            # get loss for the given partition
+            if p == 0:
+                loss_set = loss_func(est_mixes, targets, **kwargs)[:, None]
+            else:
+                loss_set = torch.cat([loss_set, loss_func(est_mixes, targets, **kwargs)[:, None]], dim=1)
```

Comment:
It's more efficient make a list and use `cat`/`stack` than concatenating repeatedly.